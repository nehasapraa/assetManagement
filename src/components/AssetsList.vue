<template>
  <b-row>
    <b-col cols="12">
      <h2>
        Asset List
        <b-link href="/#/add-asset">(Add Asset)</b-link>
      </h2>
      <b-table striped hover :items="assets" :fields="fields" :tbody-tr-class="rowClass" name="assetList">
      </b-table>
      <ul v-if="errors && errors.length">
        <li v-for="error of errors" :key="error.message">
          {{error.message}}
        </li>
      </ul>
    </b-col>
  </b-row>
</template>

<script>

import axios from 'axios'

export default {
  name: 'AssetList',
  data () {
    return {
      fields: [
        { 'asset_name': { label: 'Name', sortable: true } },
        { 'shopping_center.center_name': { label: 'Shopping Center', sortable: true } },
        { 'location_details': { label: 'Location' } }
      ],
      assets: [],
      errors: []
    }
  },
  created () {
    axios.get(process.env.API_URL + `/assets`)
      .then(response => {
        this.assets = response.data
      })
      .catch(e => {
        this.errors.push(e)
      })
  },

  methods: {
    rowClass (item, type) {
      if (!item || type !== 'row') return
      if (item.status === false) return 'table-warning'
    }
  }
}
</script>
