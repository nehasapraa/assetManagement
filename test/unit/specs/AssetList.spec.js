import { mount, shallowMount } from '@vue/test-utils';
//import Component from 'components/HelloWorld'; // name of your Vue component
import AssetsList from '../../../src/components/AssetsList'; // name of your Vue component
import { BRow, BCol, BLink, BTable } from 'bootstrap-vue';

let wrapper;

beforeEach(() => {
    wrapper = shallowMount(AssetsList, {
        stubs: {
          "b-row": BRow,
          "b-col": BCol,
          "b-table": BTable,
          "b-link": BLink
                      },
    });
});

afterEach(() => {
    wrapper.destroy();
});

describe('Assets List', () => {
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
});
