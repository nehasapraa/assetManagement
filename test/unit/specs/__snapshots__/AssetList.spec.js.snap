// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assets List renders correctly 1`] = `
<div
  class="row"
>
  <div
    class="col-12"
  >
    <h2>
      
      Asset List
      
      <a
        class=""
        href="/#/add-asset"
        target="_self"
      >
        (Add Asset)
      </a>
    </h2>
     
    <table
      aria-busy="false"
      aria-colcount="3"
      class="table b-table table-striped"
      id="__BVID__5"
      name="assetList"
      role="table"
    >
      <!---->
      <!---->
      <thead
        class=""
        role="rowgroup"
      >
        <!---->
        <tr
          class=""
          role="row"
        >
          <th
            aria-colindex="1"
            aria-sort="none"
            class=""
            role="columnheader"
            scope="col"
            tabindex="0"
          >
            Name
            <span
              class="sr-only"
            >
               (Click to sort Ascending)
            </span>
          </th>
          <th
            aria-colindex="2"
            aria-sort="none"
            class=""
            role="columnheader"
            scope="col"
            tabindex="0"
          >
            Shopping Center
            <span
              class="sr-only"
            >
               (Click to sort Ascending)
            </span>
          </th>
          <th
            aria-colindex="3"
            class=""
            role="columnheader"
            scope="col"
            tabindex="0"
          >
            Location
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      />
      <!---->
    </table>
     
    <!---->
  </div>
</div>
`;
